### DISCUSSION TITLE
WindTerm 2.2.0 Prerelease 1

### DISCUSSION THREAD

**Discussion by kingToolbox** [kingToolbox/WindTerm#204] [General]
# Usage:
### Windows
Please install **[vc_redist.x86](https://aka.ms/vs/16/release/vc_redist.x86.exe)** if WindTerm **failed** to run.
Please use **WindTerm_Win10_Conpty.exe** to support Windows 10 shell. (**Windows 19H1 [v10.0.18309] and later**)

### MacOS
Please open dmg file and drag WindTerm.app to the `Applications` folder.

### Linux
Please use the following command to run the WindTerm after downloading.
``` shell
tar -xzvf WindTerm_2.2.0_Prerelease_1_Linux_Portable.tar.gz && cd ./WindTerm_2.2.0 && chmod +x ./WindTerm && ./WindTerm
```

### Upgrade
For Windows and Linux, copy or move the **profiles** folder from the **old Windows/Linux version** to the **new Windows/Linux version** and all the sessions and configuration will be restored.

For MacOS, copy or move the **profiles** folder from the old **WindTerm.app/Contents/MacOS** to the new **WindTerm.app/Contents/MacOS** and all the sessions and configuration will be restored.

The **profiles** folder is cross-platform. 

# Note:

Starting from this version, **unless necessary or the last pre-release version**, the MacOS/Linux version will no longer be provided in the pre-release stage. 

# Known Issues:
- [ZMODEM] **Only support file names contains ASCII characters in Windows, otherwise the file name of the transfered file will be garbled after transmission, but it does not affect the correctness of the file content.** Linux and MacOS version does not have this problem.

# Changes in Version 2.2.0 Prerelease 1 (2021-09-25):
# Features:
- [SSH] **Supports SSH ControlMaster.** #173 
- [SHELL] **Added a new `Shell` pane makes it more easier to use shells, such as Bash, Cmd, Powershell, WSL, etc.**
- [TERM] **Supports `hard reset`, `soft reset`, and `character set reset` of the term.** #162 #199 
- [TERM] Supports setting the number of lines each time the mouse wheel scrolls in the alternative screen. #183 
- [GUI] Added an identity file path input box in the `Session Setting - SSH - Authentication` dialog. 
- [QUICKBAR] **Added a shortcut menu containing `Modify` and `Remove` actions to the command button to quickly manage the button.** 
- [MESSAGEBAR] Supports simultaneous display of multiple messages in expanded mode. 

# Improvements:
- [APP] **Redesigned, rewritten, and refactored more than 10,000 lines of core code to enhance flexibility, reliability and robustness.** 
- [GUI] Supports restoring the `Focus Mode` after restarting. #175 #197 
- [GUI] Supports restoring the visibility of the menu bar and status bar after restarting. #174 
- [FILER] When downloading a single file, the menu item `Open Target Directory` will select the file when opening the target folder. 
- [MESSAGEBAR] Added the time stamp of the message to the tooltip. 

# Bugs:
- [SSH] Cannot log in using an identity file in a path that contains non-English characters. #159 
- [QUICKBAR] When the `Rename Group` dialog box is canceled, the group name will be cleared. #166 #191 
- [QUICKBAR] When multiline text is sent to Windows Cmd, the carriage return of each line is not executed correctly. 
- [QUICKBAR] When no group is set to be visible, the `Quick` button cannot pop up the menu correctly. 
- [GUI] When the `Paste` dialog was initially opened, the number of characters and lines of the pasted text could not be displayed correctly. 
- [GUI] When the browse buttons of `X/Y/Z Modem - Download/Upload Directory` in the `Session Settings` dialog are clicked, the browse dialog is not displayed. 
- [GUI] When all sessions are closed, the `Local/Remote` button in the status bar is not hidden correctly. 
- [GUI] Executing `Close All Views` will cause a small amount of memory leaks. 

<hr /><em>This discussion was created from the release <a href='https://github.com/kingToolbox/WindTerm/releases/tag/2.2-prerelease'>WindTerm 2.2.0 Prerelease 1</a>.</em>

**Comment by gooker** [comment]
Start with focus mode  quick bar not show  **group**. need manual select 
<img width="334" alt="20210926131332" src="https://user-images.githubusercontent.com/5688076/134794653-7cc28044-0dba-46eb-9834-d538cc799390.png">

@kingToolbox

**Reply by kingToolbox** [reply]
I tested it and found no problems. Please make sure that quickbar is not set to `Hide` in the focuse mode dialog (MenuBar - Mode - Focuse Mode Settings). It is best to set this switch to `Show`, if it is set to `Preserve`, then the quickbar should be visible before exiting.

**Reply by kingToolbox** [reply]
I re-tested and found that there is indeed a problem with the quickbar groups display as you said, I will fix it in the next version. Thank you.

**Reply by kingToolbox** [reply]
Hi, I have fixed this issue and just released the new prerelease version [WindTerm_2.2.0_Prerelease_2](https://github.com/kingToolbox/WindTerm/releases/tag/2.2-prerelease), please download and check it, thank you.

**Reply by eugenov** [reply]
WindTerm 2.2.0 Prerelease 2, Session - Pre**p**erence typo in Menu :)

**Reply by kingToolbox** [reply]
@eugenov 
Wow, I didn't even notice. It has been revised, because a new version with multiple languages will be released soon, so the program will be updated next time. Thank you very much!

**Comment by helloxz** [comment]
Is there no macos version of WindTerm 2.2.0 Prerelease 4?

**Reply by eugenov** [reply]
As said in Note part, no 2.2.0 release for macos till next stable in december.
